ngraph/frontend/onnx_import/src/utils/arg_min_max_factory.cpp

```diff
@@ -34,19 +36,57 @@ namespace ngraph
             {
                 const auto k_node =
                     default_opset::Constant::create(ngraph::element::i64, Shape{}, {1});
-                const auto topk = std::make_shared<default_opset::TopK>(
+
+                // Reverse input for top-k
+                if (m_select_last_index == 1)
+                {
+                    const auto axis_node =
+                        default_opset::Constant::create(ngraph::element::i64, Shape{1}, {m_axis});
+                    auto reverse = std::make_shared<default_opset::Reverse>(
+                        m_input_node, axis_node, default_opset::Reverse::Mode::INDEX);
+
+                    auto topk = std::make_shared<default_opset::TopK>(
+                        reverse, k_node, m_axis, mode, default_opset::TopK::SortType::NONE);
+
+                    auto data_shape = std::make_shared<default_opset::ShapeOf>(m_input_node);
+                    auto dims_on_axis = std::make_shared<default_opset::Gather>(
+                        data_shape,
+                        axis_node,
+                        default_opset::Constant::create(ngraph::element::i64, Shape{}, {0}));
+
+                    auto res_index = std::make_shared<default_opset::Subtract>(
+                        dims_on_axis,
+                        std::make_shared<default_opset::Convert>(topk->output(1), element::i64));
+                    auto result = std::make_shared<default_opset::Subtract>(
+                        res_index,
+                        default_opset::Constant::create(ngraph::element::i64, Shape{1}, {1}));
+
+                    if (m_keep_dims == 0)
+                    {
+                        const auto axis_to_remove = default_opset::Constant::create(
+                            element::u64, Shape{}, {topk->get_axis()});
+
+                        return std::make_shared<default_opset::Squeeze>(result, axis_to_remove);
+                    }
+
+                    return result;
+                }
```

Comment:
The previous version of this code was much simpler, why a reverse mode need so much code to support it?
Could you add some comments to it?

Reply:
Indeed it needs a lot of supporting code, otherwise we need more sorting modes in TopK op.

Reply:
Done